From bdf9ecb7e2b8b22484e94001530607de65b949e8 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Fri, 18 Sep 2015 15:13:14 -0400
Subject: [PATCH 071/134] trace-cmd listen: Change install location to handle
 /usr/local

If prefix is "/usr" then use the "/var" directory, but for everything else
install in "$(prefix)/var".

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 Makefile       | 16 +++++++++++++++-
 trace-listen.c |  5 ++++-
 2 files changed, 19 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index afdeef4fec35..c058adb1e121 100644
--- a/Makefile
+++ b/Makefile
@@ -53,9 +53,16 @@ export man_dir man_dir_SQ html_install html_install_SQ INSTALL
 export img_install img_install_SQ
 export DESTDIR DESTDIR_SQ
 
+ifeq ($(prefix),/usr)
+root = /
+else
+root = $(prefix)/
+endif
+
 ifeq ($(prefix),$(HOME))
 plugin_dir = $(HOME)/.trace-cmd/plugins
 python_dir = $(HOME)/.trace-cmd/python
+var_dir = $(HOME)/.trace-cmd/
 else
 plugin_dir = $(prefix)/$(libdir)/trace-cmd/plugins
 python_dir = $(prefix)/$(libdir)/trace-cmd/python
@@ -63,13 +70,20 @@ PLUGIN_DIR = -DPLUGIN_DIR="$(plugin_dir)"
 PYTHON_DIR = -DPYTHON_DIR="$(python_dir)"
 PLUGIN_DIR_SQ = '$(subst ','\'',$(PLUGIN_DIR))'
 PYTHON_DIR_SQ = '$(subst ','\'',$(PYTHON_DIR))'
+var_dir = $(root)var
 endif
 
+VAR_DIR = -DVAR_DIR="$(var_dir)"
+VAR_DIR_SQ = '$(subst ','\'',$(VAR_DIR))'
+var_dir_SQ = '$(subst ','\'',$(var_dir))'
+
 HELP_DIR = -DHELP_DIR=$(html_install)
 HELP_DIR_SQ = '$(subst ','\'',$(HELP_DIR))'
 
 BASH_COMPLETE_DIR ?= /etc/bash_completion.d
 
+export var_dir
+
 # copy a bit from Linux kbuild
 
 ifeq ("$(origin V)", "command line")
@@ -255,7 +269,7 @@ LIBS += -laudit
 endif
 
 # Append required CFLAGS
-override CFLAGS += $(CONFIG_FLAGS) $(INCLUDES) $(PLUGIN_DIR_SQ)
+override CFLAGS += $(CONFIG_FLAGS) $(INCLUDES) $(PLUGIN_DIR_SQ) $(VAR_DIR)
 override CFLAGS += $(udis86-flags) $(blk-flags)
 
 ifeq ($(VERBOSE),1)
diff --git a/trace-listen.c b/trace-listen.c
index 1e3375f76f3d..db3a016ecc91 100644
--- a/trace-listen.c
+++ b/trace-listen.c
@@ -37,7 +37,10 @@
 
 #define MAX_OPTION_SIZE 4096
 
-#define VAR_RUN_DIR		"/var/run"
+#define _VAR_DIR_Q(dir)		#dir
+#define VAR_DIR_Q(dir)		_VAR_DIR_Q(dir)
+
+#define VAR_RUN_DIR		VAR_DIR_Q(VAR_DIR) "/run"
 
 static char *default_output_dir = ".";
 static char *output_dir;
-- 
2.5.5

